Update project to build for newer balenaOS

klutchell opened the pull request:
Some recent balenaOS images using kernel 5.10 and gcc 9.3.0
are unable to compile kernel modules using gcc 8 in debian buster.

`gcc: error: unrecognized command line option '-mstack-protector-guard=sysreg'`

Resolves: https://github.com/balena-os/kernel-module-build/issues/37
Change-type: patch
Signed-off-by: Kyle Harding <kyle@balena.io>

klutchell committed:
Install gcc 10 from debian testing

Some recent balenaOS images using kernel 5.10 and gcc 9.3.0
are unable to compile kernel modules using gcc 8 in debian buster.

`gcc: error: unrecognized command line option '-mstack-protector-guard=sysreg'`

Change-type: patch
Signed-off-by: Kyle Harding <kyle@balena.io>

klutchell committed:
Also install awscli from the testing repo

Change-type: patch
Signed-off-by: Kyle Harding <kyle@balena.io>

zwhitchcox reviewed (commented):
It looks like you're installing gcc-9 on line 6...why does this say gcc 10?

rahul-thakoor commented:
using

```dockerfile
FROM balenalib/raspberrypi4-64-debian:buster

RUN install_packages curl wget build-essential libelf-dev awscli bc flex libssl-dev python bison

# install gcc 10
RUN echo 'deb http://deb.debian.org/debian testing main' >> /etc/apt/sources.list && install_packages awscli python3-yaml gcc-9 libc6-dev

COPY . /usr/src/app
WORKDIR /usr/src/app

ENV VERSION '2.88.4.dev'
RUN BALENA_MACHINE_NAME=raspberrypi4-64 ./build.sh build --device raspberrypi4-64 --os-version "$VERSION" --src example_module

CMD [ "/usr/src/app/run.sh" ]
```

we can't install `gcc-9`. i guess this is no longer available?

right now if we try to build for 2.88.4, it works but the current base image used is bullseye imo, not buster

klutchell commented:
> It looks like you're installing gcc-9 on line 6...why does this say gcc 10?

this PR is from a year ago, I honestly don't recall what was going through my head at the time

klutchell commented:
> using
> 
> ```dockerfile
> FROM balenalib/raspberrypi4-64-debian:buster
> 
> RUN install_packages curl wget build-essential libelf-dev awscli bc flex libssl-dev python bison
> 
> # install gcc 10
> RUN echo 'deb http://deb.debian.org/debian testing main' >> /etc/apt/sources.list && install_packages awscli python3-yaml gcc-9 libc6-dev
> 
> COPY . /usr/src/app
> WORKDIR /usr/src/app
> 
> ENV VERSION '2.88.4.dev'
> RUN BALENA_MACHINE_NAME=raspberrypi4-64 ./build.sh build --device raspberrypi4-64 --os-version "$VERSION" --src example_module
> 
> CMD [ "/usr/src/app/run.sh" ]
> ```
> 
> we can't install `gcc-9`. i guess this is no longer available?
> 
> right now if we try to build for 2.88.4, it works but the current base image used is bullseye imo, not buster

@rahul-thakoor Yeah, we should have pinned to buster as part of this PR. If you've confirmed it works can you push the additional required commit(s)?

rahul-thakoor committed:
Update project to build for newer balenaOS

- pin to bullseye
- use debian-python base image
- tested with balenaOS v2.88.4

Change-type: major
Signed-off-by: Rahul Thakoor <rahul.thakoor@gmail.com>

rahul-thakoor commented:
@klutchell can you please check?

I couldn't get it to work with buster. i pinned to bullseye

alexgg reviewed (approved):
lgtm
